airflow/migrations/versions/0152_2_10_3_fix_dag_schedule_dataset_alias_reference_naming.py

```diff
@@ -0,0 +1,123 @@
+    *,
+    batch_op: BatchOperations,
+    original_name: str | conv,
+    new_name: str | conv,
+    referent_table: str,
+    local_cols: list[str],
+    remote_cols: list[str],
+    ondelete: str,
+) -> None:
+    batch_op.drop_constraint(original_name, type_="foreignkey")
+    batch_op.create_foreign_key(
+        constraint_name=new_name,
+        referent_table=referent_table,
+        local_cols=local_cols,
+        remote_cols=remote_cols,
+        ondelete=ondelete,
+    )
+
+
+def upgrade():
+    """Rename dag_schedule_dataset_alias_reference constraint."""
+    with op.batch_alter_table("dag_schedule_dataset_alias_reference", schema=None) as batch_op:
+        bind = op.get_context().bind
+        insp = inspect(bind)
+        fk_constraints = [fk["name"] for fk in insp.get_foreign_keys("dag_schedule_dataset_alias_reference")]
+
+        # "dsdar_dataset_alias_fkey" was the constraint name defined in the model while "dsdar_dataset_fkey" is the one
+        # defined in the previous migration.
+        # Rename this constraint name if user is using the name "dsdar_dataset_fkey".
+        if "dsdar_dataset_fkey" in fk_constraints:
+            _rename_fk_constraint(
+                batch_op=batch_op,
+                original_name="dsdar_dataset_fkey",
+                new_name="dsdar_dataset_alias_fkey",
+                referent_table="dataset_alias",
+                local_cols=["alias_id"],
+                remote_cols=["id"],
+                ondelete="CASCADE",
+            )
+
+        # "dsdar_dag_fkey" was the constraint name defined in the model while "dsdar_dag_id_fkey" is the one
+        # defined in the previous migration.
+        # Rename this constraint name if user is using the name "dsdar_dag_fkey".
+        if "dsdar_dag_fkey" in fk_constraints:
+            _rename_fk_constraint(
+                batch_op=batch_op,
+                original_name="dsdar_dag_fkey",
+                new_name="dsdar_dag_id_fkey",
+                referent_table="dataset_alias",
+                local_cols=["alias_id"],
+                remote_cols=["id"],
+                ondelete="CASCADE",
+            )
+
+
+def downgrade():
+    """Undo dag_schedule_dataset_alias_reference constraint rename."""
+    with op.batch_alter_table("dag_schedule_dataset_alias_reference", schema=None) as batch_op:
+        _rename_fk_constraint(
+            batch_op=batch_op,
+            original_name="dsdar_dataset_alias_fkey",
+            new_name="dsdar_dataset_fkey",
+            referent_table="dataset_alias",
+            local_cols=["alias_id"],
+            remote_cols=["id"],
+            ondelete="CASCADE",
+        )
+        _rename_fk_constraint(
+            batch_op=batch_op,
+            original_name="dsdar_dag_id_fkey",
+            new_name="dsdar_dag_fkey",
+            referent_table="dataset_alias",
+            local_cols=["alias_id"],
+            remote_cols=["id"],
+            ondelete="CASCADE",
+        )
```

Comment:
You will need same "if" condition so it doesn't try to drop a constraint that doesn't exist

Reply:
I can't think of a case that it won't exist 🤔 but no harm to safe guard. will update it